packages/noya-colorpicker/src/components/GradientSlider.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useCallback, useRef } from 'react';
+import styled from 'styled-components';
+import { useEventCallback } from '../hooks/useEventCallback';
+import { useIsomorphicLayoutEffect } from '../hooks/useIsomorphicLayoutEffect';
+import { clamp } from '../utils/clamp';
+
+const Container = styled.div(() => ({
+  position: 'absolute' as any,
+  left: 0,
+  top: 0,
+  right: 0,
+  bottom: 0,
+  borderRadius: 'inherit',
+  outline: 'none',
+  /* Don't trigger the default scrolling behavior when the event is originating from this element */
+  touchAction: 'none',
+}));
+
+export interface Interaction {
+  left: number;
+  top: number;
+}
+
+// Check if an event was triggered by touch
+const isTouch = (e: MouseEvent | TouchEvent): e is TouchEvent => 'touches' in e;
+
+// Returns a relative position of the pointer inside the node's bounding box
+const getRelativePosition = (
+  node: HTMLDivElement,
+  event: MouseEvent | TouchEvent,
+): Interaction => {
+  const rect = node.getBoundingClientRect();
+
+  // Get user's pointer position from `touches` array if it's a `TouchEvent`
+  const pointer = isTouch(event) ? event.touches[0] : (event as MouseEvent);
+
+  return {
+    left: clamp(
+      (pointer.pageX - (rect.left + window.pageXOffset)) / rect.width,
+    ),
+    top: clamp((pointer.pageY - (rect.top + window.pageYOffset)) / rect.height),
+  };
+};
+
+interface Props {
+  onMove: (interaction: Interaction) => void;
+  onMoveChange: (moving: boolean) => void;
+  onClick: (interaction: Interaction) => void;
+  onKey: (offset: Interaction) => void;
+  isOnPoint: (interaction: Interaction) => boolean;
+  children: React.ReactNode;
+}
+
+const InteractiveBase = ({
```

Comment:
Since this file is kinda complex and still mostly the same as `Interactive`, it'd be nice if we could share some/all of the logic. What's unique about this version? I'm wondering if we can make it an optional prop?

Reply:
I'll try combining them and doing the new onClick event 🤔 Maybe they can just work together